crates/bevy_reflect/bevy_reflect_derive/src/impls/type_name.rs

```diff
@@ -0,0 +1,80 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::quote;
+use syn::{Generics, Path};
+
+pub(crate) fn impl_type_name(
+    type_name: &Ident,
+    generics: &Generics,
+    reflected_type_name: Option<&str>,
+    bevy_reflect_path: &Path,
+) -> proc_macro2::TokenStream {
+    let is_generic = !generics.params.is_empty();
+
+    let base_name = reflected_type_name
+        .map(|x| quote!(#x))
+        .unwrap_or_else(|| quote!(concat!(module_path!(), "::", stringify!(#type_name))));
```

Comment:
Nit: I saw on the docs for `stringify!` that it says: 

>Note that the expanded results of the input tokens may change in the future. You should be careful if you rely on the output.

I don't know what this means haha, but if you think it's cause for concern, we could just do something like:

```rust
let string_name = type_name.to_string();
// ...
  .unwrap_or_else(|| quote!(concat!(module_path!(), "::", #string_name)));
```

Reply:
It basically means that it for example is allowed to add or remove whitespace if it would result in the same code when parsing. I like your suggestion.